outbox-event-publisher/src/main/java/org/jvalue/outboxeventpublisher/OutboxTableTransform.java

```diff
@@ -0,0 +1,98 @@
+package org.jvalue.outboxeventpublisher;
+
+import io.debezium.config.Configuration;
+import io.debezium.data.Envelope;
+import io.debezium.transforms.SmtManager;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.kafka.common.config.ConfigDef;
+import org.apache.kafka.connect.connector.ConnectRecord;
+import org.apache.kafka.connect.errors.ConnectException;
+import org.apache.kafka.connect.transforms.ExtractField;
+import org.apache.kafka.connect.transforms.Transformation;
+
+import java.util.Map;
+
+import static org.apache.kafka.connect.transforms.util.Requirements.requireStruct;
+
+@Slf4j
+public class OutboxTableTransform<R extends ConnectRecord<R>> implements Transformation<R> {
+  private final ExtractField<R> afterExtractor = new ExtractField.Value<>();
+
+  private String fieldEventId;
+  private String fieldEventRoutingKey;
+  private String fieldEventPayload;
+
+  private SmtManager<R> smtManager;
+
+  @Override
+  public R apply(R r) {
+    // Ignoring tombstones
+    if (r.value() == null) {
+      log.debug("Tombstone message ignored. Message key: \"{}\"", r.key());
+      return null;
+    }
+
+    // Ignoring messages which do not adhere to the CDC Envelope, for instance:
+    // Heartbeat and Schema Change messages
+    if (!smtManager.isValidEnvelope(r)) {
+      return r;
```

Comment:
Ignoring meant in all other cases to return null, why is it different here?

Reply:
I have copied this from an outbox event transformation for Kafka which debezium provides out of the box. I think for us it does make sense to also return null, because we do not handle heartbeat and schema change messages.

Reply:
Please make sure to not copy code from the internet since that will potentially get us into copyright issues! Although it might seem stupid, please remove the copied stuff and write the part by yourself in a new commit...

Reply:
Done and force pushed everything

Reply:
Perfect, thanks a lot! Sorry for the inconveniences..